x-pack/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {
+  Position,
+  Chart,
+  Settings,
+  Axis,
+  LineSeries,
+  getAxisId,
+  getSpecId,
+  AreaSeries,
+  BarSeries,
+} from '@elastic/charts';
+import { Datatable } from '../../../canvas/canvas_plugin_src/functions/types';
+
+/**
+ * This file contains TypeScript type definitions and their equivalent expression
+ * definitions, for configuring and rendering an XY chart. The XY chart serves
+ * triple duty as a bar, line, or area chart.
+ *
+ * The xy_chart expression function serves mostly as a passthrough to the xy_chart_renderer
+ * which does the heavy-lifting.
+ */
+
+/**
+ * Configuration for the chart's legend.
+ */
+export interface LegendConfig {
+  isVisible: boolean;
+  position: Position;
+}
+
+/**
+ * LegendConfig as an expression type.
+ */
+export const legendConfig = {
+  name: 'legendConfig',
+  aliases: [],
+  type: 'legendConfig',
+  help: `Configure a chart's legend`,
+  context: {
+    types: ['null'],
+  },
+  args: {
+    isVisible: { types: ['boolean'] },
+    position: { types: ['string'] },
```

Comment:
Looks like you're not taking advantage of having either default values or restricted options in the expression definition here. See `axisConfig.ts`

Reply:
That's true. Defaults were on my TODO list for a later PR, but I'm OK with adding them now.